### DISCUSSION TITLE
How to update Motion from an other build version

### DISCUSSION THREAD

**Discussion by titithebad** [Motion-Project/motion#1807] [General]
Hello,
I have been using Motion for a long time on other debian versions on my raspberries (Buster and Bullseye). I see that updates are arriving each time on the latest version of debian (currently bookworm). How do I use the latest version of Motion on an older version of Debian? I saw that it was possible to recompile from the code but that sometimes it didn't work very well due to versions of certain libraries.
Regards !

**Comment by Mr-Dave** [comment]
Generally, building on older debian versions works without any issue.  Library problems sound like you've tried to install a deb package from a different version which won't work at all and could ruin many things is you try to force it.  The following are the steps.

1.  Make sure system is up to date with all the other packages.  i.e. Keep doing `sudo apt update` `sudo apt dist-upgrade` `sudo apt autoremove` until the system reports everything is current.  Do not proceed to the next step until this is all "clean"
2. Use the [builddeb](https://motion-project.github.io/motion_build.html#Build_Deb) script to create a new deb.  The script will indicate what additional packages need to be installed.  i.e. `./builddeb.sh YourName Anyaddress@nowhere.com 4.7`
3. Then you can install the deb via `dpkg` (since you know you have all the packages installed) or `gdebi`

**Reply by titithebad** [reply]
Ok, I'm trying this weekend and I'll tell you !
